fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java

```diff
@@ -1436,6 +1471,210 @@ public TSetConfigResponse setConfig(TSetConfigRequest request) throws TException
+
+        if (request.partitionValues == null) {
+            result.setStatus(errorStatus);
+            result.setErr_msg("partitionValues should not null.");
+            return result;
+        }
+        // Now only supports the case of automatically creating single partition
+        if (request.partitionValues.size() != 1) {
+            result.setStatus(errorStatus);
+            result.setErr_msg("only support single partition, partitionValues size should equal 1.");
+            return result;
+        }
+        List<String> partitionValues = request.partitionValues.get(0);
+
+        Set<String> partitionNames = Sets.newHashSet();
+        PartitionInfo partitionInfo = olapTable.getPartitionInfo();
+        if (!(partitionInfo instanceof ExpressionRangePartitionInfo)) {
+            result.setStatus(errorStatus);
+            result.setErr_msg("only support expression range partition.");
+            return result;
+        }
+        List<Expr> partitionExprs = ((ExpressionRangePartitionInfo) partitionInfo).getPartitionExprs();
+        if (partitionExprs.size() != 1) {
+            result.setStatus(errorStatus);
+            result.setErr_msg("only support one expression partitionExpr.");
+            return result;
+        }
+        Expr expr = partitionExprs.get(0);
+        if (!(expr instanceof FunctionCallExpr)) {
+            result.setStatus(errorStatus);
+            result.setErr_msg("only support FunctionCallExpr");
+            return result;
+        }
+        FunctionCallExpr functionCallExpr = (FunctionCallExpr) expr;
+        String fnName = functionCallExpr.getFnName().getFunction();
+        if (!fnName.equals(FunctionSet.DATE_TRUNC)) {
+            result.setStatus(errorStatus);
+            result.setErr_msg("only support data_trunc function.");
+            return result;
+        }
+        List<Expr> paramsExprs = functionCallExpr.getParams().exprs();
+        if (paramsExprs.size() != 2) {
+            result.setStatus(errorStatus);
+            result.setErr_msg("params exprs size should be 2.");
+            return result;
+        }
+        Expr granularityExpr = paramsExprs.get(0);
+        if (!(granularityExpr instanceof StringLiteral)) {
+            result.setStatus(errorStatus);
+            result.setErr_msg("granularity is not string literal.");
+            return result;
+        }
+        StringLiteral granularityLiteral = (StringLiteral) granularityExpr;
+        String granularity = granularityLiteral.getStringValue();
+        for (String partitionValue : partitionValues) {
+            DateTimeFormatter beginDateTimeFormat;
+            LocalDateTime beginTime;
+            LocalDateTime endTime;
+
+            try {
+                beginDateTimeFormat = DateUtils.probeFormat(partitionValue);
+                beginTime = DateUtils.parseStringWithDefaultHSM(partitionValue, beginDateTimeFormat);
+                switch (granularity.toLowerCase()) {
```

Comment:
Should the data parsing code be extracting into a Util and test it well ?

Reply:
You are right